scene/main/viewport.cpp

```diff
@@ -1838,8 +1840,30 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 		if (gui.drag_data.get_type() == Variant::NIL && over && !gui.modal_stack.empty()) {
 
 			Control *top = gui.modal_stack.back()->get();
+
 			if (over != top && !top->is_a_parent_of(over)) {
-				over = NULL; //nothing can be found outside the modal stack
+
```

Comment:
Can you add a bit of comment here to explain that this code is used for adding a more natural feeling to menu buttons?

Reply:
Comment added.